warehouse/query-core/src/test/java/datawave/query/iterator/logic/NegationFilterTest.java

```diff
@@ -176,5 +177,49 @@ public boolean isContextRequired() {
         public void setContext(K context) {
             // no-op
         }
+
+        // this iterator shouldn't need to use the index only flag
```

Comment:
index only != nonEvent

Reply:
there are places where we've conflated these terms and caused bugs.  Double check which you mean and make methods and comments consistent